src/MeshTransfer.f90

```diff
@@ -0,0 +1,202 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief A Fortran 2003 module implementing some basic sorting algorithms.
+!>
+!> @par Module Dependencies
+!>  - @ref IntrType "IntrType": @copybrief IntrType
+!>
+!> @author Ben Collins
+!>    @date 12/22/2017
+!>
+!> @par Revisions:
+!>   (07/06/2014) - Benjamin Collins
+!>   - Added additional interface for qsort (QuickSort) to give better
+!>           performance when needed
+!>   (3/16/2016) - Dan Jabaay
+!>   - Added sorting of one array based on the sorting of another.  This
+!>     functionality will help with more applicable sorting cases.
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+MODULE MeshTransfer
+#include "DBC.h"
+  USE DBC
+  USE IntrType
+  USE Strings
+  IMPLICIT NONE
+
+  PUBLIC :: LPPointVal
+  PUBLIC :: LPIntegral
+  PUBLIC :: ZPPointVal
+  PUBLIC :: ZPIntegral
+
+  REAL(SRK),PARAMETER :: ONE=1.0_SRK
+  REAL(SRK),PARAMETER :: TWO=2.0_SRK
+
+!
+!===============================================================================
+  CONTAINS
+!
+!-------------------------------------------------------------------------------
+!> @brief Returns value of an arbitrary order Legendre Polynomial with coefficients coef
+!> @param coef  The coefficients of the Legendre expansion
+!> @param globX The value of x in the global coordinate
+!> @param h     The height of the node, OPTIONAL (if not provided a height of 2 is assumed)
+!> @param nodeX The center of the node, OPTIONAL (if not provided a position of 0 is assumed)
+!>
+!> This routine returns the value of an arbitrary order Legendre expansion at a
+!> specific point for NEM and SANM nodal methods.
+!>
+    FUNCTION LPPointVal(coef,globX,h,nodeX) RESULT(y)
+      REAL(SRK),DIMENSION(:),INTENT(IN) :: coef
```

Comment:
Less characters and better column orientation if you do:
```REAL(SRK),INTENT(IN) :: coef(:)```

Reply:
This was here for a long time.  I just moved the code.  I'll change it to your way because I like it more.